client/src/components/forms/CreateAccount.jsx

```diff
@@ -0,0 +1,240 @@
+    "Economics & Finance",
+    "Electrical & Electronics Engineering",
+    "Humanities",
+    "Mathematics",
+    "Mechanical Engineering",
+    "Pharmacy",
+    "Physics",
+];
+const CreateAccount = (props) => {
+    const [open, setOpen] = useState({
+        isOpen:  props['open'],
+        username: "",
+        password: "",
+        name: "",
+        email: "",
+        confPass: "",
+        existingUsername: false,
+        weakPassword: false,
+        matchPass: true,
+        emptyUserField: true,
+        emptyPassField: true,
+        emptyNameField: true,
+        emptyDeptField: true,
+        emptyEmailField: true,
+    });
+   
+    const editUsername = (e) => {
+        e.preventDefault();
+        setOpen({
+            ...open,
+            username: e.target.value,
+            existingUsername: false,
+            emptyUserField: (e.target.value === null || e.target.value.length === 0) ? true : false
+        });
+    }
+    const editPassword = (e) => {
+        e.preventDefault();
+        setOpen({
+            ...open,
+            password: e.target.value,
+            weakPassword: (e.target.value.length < 8) ? true : false,
+            emptyPassField: (e.target.value === null || e.target.value.length === 0) ? true : false
+        });
+    }
+    const editConf = (e) => {
+        const eq = (e.target.value === open.password);
+        e.preventDefault();
+        setOpen({
+            ...open,
+            confPass: e.target.value,
+            matchPass: (eq) ? true : false,
+            emptyPassField: (e.target.value === null || e.target.value.length === 0) ? true : false
+        });
+    }
+    const editName = (e) => {
+        e.preventDefault();
+        setOpen({
+            ...open,
+            name: e.target.value,
+            emptyNameField: (e.target.value === null || e.target.value.length === 0) ? true : false
+        })
+    }
+    const editEmail = (e) => {
+        e.preventDefault();
+        setOpen({
+            ...open,
+            email: e.target.value,
+            emptyEmailField: (e.target.value === null || e.target.value.length === 0) ? true : false,
+            invalidEmail: (e.target.value.match("^([a-zA-Z0-9_\\-\\.]+)@([a-zA-Z0-9_\\-\\.]+)\\.([a-zA-Z]{2,5})$") === null) ? true : false,
+        })
+    }
+    const editDept = (e) => {
+        e.preventDefault();
+        setOpen({
+            ...open,
+            department: e.target.value,
+            emptyDeptField: e.target.value === null ? true : false
+        })
+    }
+    const submit = (e) => {
+        let iu = false, io = true;
+        e.preventDefault();
+        if (open.confPass === open.password
+            && !open.invalidEmail
+            && !open.weakPassword
+            && open.matchPass
+            && !open.emptyDeptField
+            && !open.emptyEmailField
+            && !open.emptyNameField
+            && !open.emptyUserField
+            && !open.emptyPassField
```

Comment:
Maybe just destructure the state at the beginning of the file so you can directly access `invalidEmail`, `weakPassword` etc.

Reply:
right will do

Reply:
well it's really not much of a change lol. But okay.
Is there a better way to store this? maybe bitmask?
But idt that'll be worth it anyways